force_wfmanager/ui/setup/process/data_source_view.py

```diff
@@ -42,56 +39,28 @@ class TableRow(HasStrictTraits):
 
 
 class InputSlotRow(TableRow):
-    """Row in the UI representing DataSource inputs. """
-
-    # ------------------
-    # Derived Attributes
-    # ------------------
-
-    #: Name of the slot.
-    name = Identifier()
+    """Row in the UI representing DataSource input_slots. """
 
     # -------------------
-    #     Listeners
+    # Required Attributes
     # -------------------
 
-    @on_trait_change('model.input_slot_info[]')
-    def update_view(self):
-        """Synchronises the InputSlotRow with the underlying model"""
-        self.name = self.model.input_slot_info[self.index].name
+    #: Model of the evaluator
+    model = Instance(InputSlotInfo)
 
-    @on_trait_change('name')
-    def update_model(self):
-        """Synchronises the model if the user changes a model object's name
-        via the UI
-        """
-        self.model.input_slot_info[self.index].name = self.name
+    # Available names for input variables
+    available_names = List(Identifier)
```

Comment:
At the moment this is unused, but could be useful in the future if allowing the user to select between existing variable names